docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+### Using Environment Variables
+
+In order to make your configuration usable by others, certain settings inside it will need to change. Fortunately Telegraf makes this easy by looking for variables in your configuration and replacing them with [environment variables](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#environment-variables). Then users can run your Telegraf configuration unmodified by simply setting those values in their environment before running it.
+
+The main settings you'll want to replace with variables are in the `[[outputs.influxdb_v2]]` section of your configuration:
+
+```
+urls = ["$INFLUX_HOST"]
+
+token = "$INFLUX_TOKEN"
+
+organization = "$INFLUX_ORG"
+```
+
+If your template allows the user to choose their own `bucket` values, you should make this a variable in your configuration also:
+
+```
+bucket = "$INFLUX_BUCKET"
+```
+
+Depending on what input plugins you use, there will likely be others that you want to make variables, such as host names, databases, or authentication credentials.
+
+> **Note:** Be sure to document all of these variables in the `README.md` for your Template so that users will know what they need to set and what values they should use.
+
+### Telegraf Configuration Labels
+
+Sometimes, it's handy to group Telegraf connfigurations by specific plugins. For example, if I want to make a change to an input plugin, being able to quickly find all the Telegraf configurations using that plugin is helpful. To this end, it's useful to add Labels such as `inputs.plugin_name` to your Telegraf configurations to help users filter in the UI. 
+
+### Adding Telegraf to your Template
+
+To add a Telegraf config file to your template, you will need to edit the template file manually using the following steps:
+
+ 1. Export your template using the command in the [submitting a template](submit_a_template.md) doc.
+   
+ 2. Add the following to the end of the file, being sure to give a name for your Telegraf configuration:
+    ```
+        ---
+        apiVersion: influxdata.com/v2alpha1
+        kind: Telegraf
+        metadata:
+            name: The Name Of Your Configuration
+            config: |
+        ```
+        
+ 3. Copy/paste the content of your configuration file into the lines below what you just added, indenting it all until it is 4 spaces further indented than the `config:` line.
+    
+
+## Testing Your Template
+
+After you export your finished Template, you should always test it to make sure that it applies cleanly, has everything you wanted to include, and that the instructions your provide in your `README.md` have all the steps needed to use it.
```

Comment:
After you export your finished ~Template~ **template**, ~you should always~ test it to make sure that it applies cleanly, has everything you ~wanted~ want to include, and that the instructions your provide in your `README.md` have all the **necessary** steps ~needed to use it~.